standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -255,7 +303,9 @@ public CompactionInfo findNextToCompact(FindNextCompactRequest rqst) throws Meta
         StringBuilder sb = new StringBuilder();
         sb.append("SELECT \"CQ_ID\", \"CQ_DATABASE\", \"CQ_TABLE\", \"CQ_PARTITION\", " +
           "\"CQ_TYPE\", \"CQ_POOL_NAME\", \"CQ_NUMBER_OF_BUCKETS\", \"CQ_ORDER_BY\", " +
-            "\"CQ_TBLPROPERTIES\" FROM \"COMPACTION_QUEUE\" WHERE \"CQ_STATE\" = '" + INITIATED_STATE + "' AND ");
+          "\"CQ_TBLPROPERTIES\" FROM \"COMPACTION_QUEUE\" " +
+          "WHERE \"CQ_TYPE\" != " + quoteChar(TxnStore.ABORT_CLEANUP_TYPE) +
```

Comment:
why do we need this? ABORT_CLEANUP_TYPE could be only in READY_FOR_CLEANING state

Reply:
is it correct that the type for legacy aborts would be either MINOR or MAJOR?

Reply:
Yes you are right. Removed it. Done.

> is it correct that the type for legacy aborts would be either MINOR or MAJOR?
Yes legacy aborts have only these 2 types.